src/lib/apiTypesV2/Subscription.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+#define SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <string>
+#include <vector>
+
+#include "../ngsi/Duration.h"
+#include "../ngsi/Throttling.h"
+
+struct EntID
+{
+  std::string id;
+  std::string idPattern;
+  std::string type;
+  std::string toJson();
+};
+
+struct Notification
+{
+  std::vector<std::string> attributes;
+  std::string callback;
+  Throttling throttling;
+  std::string toJson();
+};
+
+struct Condition
+{
+  std::vector<std::string> attributes;
+  struct {
+    std::string q;
+    std::string geometry;
+    std::string coords;
+  } expression;
+  std::string toJson();
+};
+
+struct Subject
+{
+  std::vector<EntID> entities;
+  Condition condition;
+  std::string toJson();
+};
+
+struct Subscription
+{
+  std::string id;
+  Subject subject;
+  Duration duration;
+  Notification notification;
+  std::string toJson();
+};
```

Comment:
In NGSIv2 we have the following operations related with subscriptions that in principle will be using this class:

```
GET	/v2/subscriptions
GET	/v2/subscriptions/<subId>
POST /v2/subscriptions
PATCH /v2/subscriptions/<subId>
```

The first and the second correspond to know operations at mongoBackend, but how to do in the third and fourth case is not clear yet:

* POST /v2/subscriptions will probably reuse the existing mongoSubscribeContext operation at mongoBackend, which is based in a different subscription class (so some v2 to v1 conversion would be needed).
* I don't think that PATCH /v2/subscriptions could reuse directly existing mongoUpdateSubscription operation in mongoBackend, as the PATCH semantics are not exactly the same than the NGSIv1 subscription update.

However, in the avove scenarios don't see the need of using for example Subject outside the Subscription. Thus, probably we could stay as it is now (only one file) and split if we find we need in the future. Makes sense?